src/server/game/Spells/Spell.cpp

```diff
@@ -9012,7 +9012,7 @@ namespace Acore
             if (!target->ToGameObject()->IsInRange(_position->GetPositionX(), _position->GetPositionY(), _position->GetPositionZ(), _range))
                 return false;
         }
-        else if (!target->IsWithinDist3d(_position, _range))
+        else if (!target->IsInDist(_position, _range))
```

Comment:
was it really necessary? Normally this shouldn't change anything, as IsWithin3d simply calls IsInDist by adding GetCombatReach to the range. But GetCombatReach is normally only overridden in Unit.cpp and should be equal to 0.0f  here (default value)

Reply:
IsWithinDist3d adds GetObjectSize which has no overrides anywhere

https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/game/Entities/Object/Object.cpp#L1295

Reply:
Yes, I remembered TC, but I didn't necessarily think that the function was different on AC (and it's normal because it's an old version of TC). In researching the reason for the change I discovered that the outcome appears to be much more complex. I'm not saying that TC's fix is necessarily the right one, but I don't think that simply replacing "IsWithinDist3d" by "IsInDist"

source: https://github.com/TrinityCore/TrinityCore/pull/16290 and https://github.com/WoWManiaUK/Redemption/issues/3061

Reply:
Yes, all the functions around Distance can use a refactor